src/test/java/com/jnape/palatable/lambdakoans/AboutApplicatives.java

```diff
@@ -0,0 +1,157 @@
+import static testsupport.matchers.IterableMatcher.iterates;
+
+public class AboutApplicatives {
+
+    @Test
+    public void applicativesZipEmbeddedValuesWithEmbeddedFunctions() {
+        Fn1<Integer, Integer> inc = x -> x + 1;
+
+        assertThat(just(9).zip(just(inc)), equalTo(just(10)));
+        assertThat(just(15).zip(just(inc)), equalTo(just(__)));
+        // Explicit types help the compiler
+        assertThat(Maybe.<Integer>nothing().zip(just(inc)), equalTo(__));
+
+        Fn1<? super String, ? extends Integer> getStringLength = String::length;
+        assertThat(right("Hello").zip(right(getStringLength)), equalTo(__));
+        assertThat(right("World!").zip(right(getStringLength)), equalTo(__));
+        assertThat(Either.<String, String>left("whoops").zip(right(getStringLength)), equalTo(left(__)));
+
+        // Moving on to LambdaIterables, where "zipping" is more obvious
+        LambdaIterable<Integer> oneThroughThree = LambdaIterable.wrap(asList(1, 2, 3));
+        Applicative<Fn1<? super Integer, ? extends Integer>, LambdaIterable<?>> wrappedInc = LambdaIterable.wrap(asList(inc));
+        assertThat(oneThroughThree.zip(wrappedInc).unwrap(), iterates(2, 3, 4));
+
+        Fn1<Integer, Integer> dec = x -> x - 1;
+        LambdaIterable<Fn1<? super Integer, ? extends Integer>> twoFunctions = LambdaIterable.wrap(asList(inc, dec));
+        assertThat(oneThroughThree.zip(twoFunctions).unwrap(), iterates(2, 0, 3, 1, 4, 2));
+
+        Fn1<Integer, Integer> times3 = x -> x * 3;
+        LambdaIterable<Fn1<? super Integer, ? extends Integer>> allFunctions = LambdaIterable.wrap(asList(inc, dec, times3));
+        assertThat(oneThroughThree.zip(allFunctions).unwrap(), iterates(__()));
+    }
+
+    @Test
+    public void lazyApplicatives() {
+        // Zipping LambdaIterables is lazy because LambdaIterables are lazy
+        LambdaIterable<Integer> infiniteOnes = LambdaIterable.wrap(repeat(1));
+        Fn1<Integer, Integer> inc = x -> x + 1;
+
+        LambdaIterable<Fn1<? super Integer, ? extends Integer>> wrappedInc = LambdaIterable.wrap(asList(inc));
+        LambdaIterable<Integer> zippedOnes = infiniteOnes.zip(wrappedInc);
+        assertThat(take(3, zippedOnes.unwrap()), iterates(__()));
+
+        // We might lazily get a mapping function...
+        AtomicInteger computed = new AtomicInteger(0);
+
+        Fn1<Integer, Maybe<Fn1<? super Integer, ? extends String>>> expensiveWayToGetMaybeToString = fn1((Integer n) -> {
+            for (int i = 0; i < n; i++) {
+                computed.incrementAndGet();
+            }
+            return just(Object::toString);
+        });
+
+        Lazy<Maybe<Fn1<? super Integer, ? extends String>>> lazyGetToString = lazy(() ->
+                expensiveWayToGetMaybeToString.apply(100_000_000));
+
+        // ...then apply it with lazyZip
+        Maybe<Integer> nothing = nothing();
+        Lazy<Maybe<String>> lazyNothingToString = nothing.lazyZip(lazyGetToString);
```

Comment:
We should point out that `Maybe` is strict and that's what makes it different from the iterable case.